jupyterhub/objects.py

```diff
@@ -115,12 +115,13 @@ def bind_url(self):
             return self.url.replace(self._connect_ip, self.ip or '*', 1)
         return self.url
 
+    @gen.coroutine
     def wait_up(self, timeout=10, http=False):
         """Wait for this server to come up"""
         if http:
-            return wait_for_http_server(self.url, timeout=timeout)
+            return (yield wait_for_http_server(self.url, timeout=timeout))
```

Comment:
Why this change? It adds an additional async layer that shouldn't be needed if we return the underlying result directly.